plugins/nodes/src/js/components/NodesTable-new.tsx

```diff
@@ -0,0 +1,316 @@
+import * as React from "react";
+import { Table, Column } from "@dcos/ui-kit";
+
+import NodesList from "#SRC/js/structs/NodesList";
+import Node from "#SRC/js/structs/Node";
+
+import { SortableColumnHeader } from "ui-kit-stage/SortableColumnHeader";
+import { SortDirection } from "../types/SortDirection";
+
+import {
+  hostnameSorter,
+  hostnameRenderer,
+  hostnameSizer
+} from "../columns/NodesTableHostnameColumn";
+import {
+  regionSorter,
+  regionRenderer,
+  regionSizer
+} from "../columns/NodesTableRegionColumn";
+import {
+  zoneSorter,
+  zoneRenderer,
+  zoneSizer
+} from "../columns/NodesTableZoneColumn";
+import {
+  healthSorter,
+  healthRenderer,
+  healthSizer
+} from "../columns/NodesTableHealthColumn";
+import {
+  tasksSorter,
+  tasksRenderer,
+  tasksSizer
+} from "../columns/NodesTableTasksColumn";
+import { cpubarRenderer, cpubarSizer } from "../columns/NodesTableCPUBarColumn";
+import {
+  cpuSorter,
+  cpuRenderer,
+  cpuSizer
+} from "../columns/NodesTableCPUColumn";
+import { membarRenderer, membarSizer } from "../columns/NodesTableMemBarColumn";
+import {
+  memSorter,
+  memRenderer,
+  memSizer
+} from "../columns/NodesTableMemColumn";
+import {
+  diskbarRenderer,
+  diskbarSizer
+} from "../columns/NodesTableDiskBarColumn";
+import {
+  diskSorter,
+  diskRenderer,
+  diskSizer
+} from "../columns/NodesTableDiskColumn";
+import { gpubarSizer, gpubarRenderer } from "../columns/NodesTableGPUBarColumn";
+import {
+  gpuSorter,
+  gpuRenderer,
+  gpuSizer
+} from "../columns/NodesTableGPUColumn";
+
+interface NodesTableProps {
+  hosts: NodesList;
+  nodeHealthResponse: boolean;
+  masterRegion: string;
+}
+
+interface NodesTableState {
+  data: Node[];
+  sortDirection: SortDirection;
+  sortColumn: string;
+}
+
+type SortFunction<T> = (data: T[], sortDirection: SortDirection) => T[];
+
+// This whole component (except `render()`) will basically go into data layer.
```

Comment:
Can we remove this comment? I think it would be better placed either in the PR description or in the commit body

Reply:
maybe @nLight can remove it while working on the build issue, otherwise I'd like to get it merged as is - we need to make some progress here 💯